File: Programmers/Python/Level 2/피보나치 수/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12945


def solution(n):
    if n < 2:
        return n
    dp = [0, 1]
    for idx in range(2, n + 1):
        dp.append((dp[idx - 1] + dp[idx - 2]) % 1234567)
    return dp[n]

File: Programmers/Javascript/Level 2/피보나치 수/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12945?language=javascript

function solution(n) {
  if (n <= 1) {
    return n;
  }
  const dp = new Array(n + 1);
  (dp[0] = 0), (dp[1] = 1);

  for (let idx = 2; idx <= n; idx++) {
    dp[idx] = (dp[idx - 1] + dp[idx - 2]) % 1234567;
  }
  return dp[n];
}
